tests/jsnums-test/jsnums-test.js

```diff
@@ -44,22 +44,84 @@ R(["pyret-base/js/js-numbers"], function(JN) {
       return false;
     }
   }
-  describe("check exceptions in js-numbers methods that can't be tested in Pyret", function() {
+  describe("check functions that don't allow testing via Pyret programs", function() {
+
+    it("fromString", function() {
+      expect(JN.fromString("5", sampleErrorBacks)).toEqual(5);
+
+      var bigIntStr = "1" + new Array(309 + 1).join("0"); // 1 followed by 309 0s
+      expect(JN.fromString(bigIntStr, sampleErrorBacks)).toEqual(JN.makeBignum(bigIntStr));
+
+      // console.log(JN.fromString("1e1", sampleErrorBacks));
+      // console.log(JN.fromString("10", sampleErrorBacks));
+      // console.log(JN.makeBignum("1e1", sampleErrorBacks));
+      // console.log(JN.makeBignum("10", sampleErrorBacks).toFixnum());
+
+      expect(JN.fromString("1e1", sampleErrorBacks)).toBe(10);
+      expect(JN.fromString("1e30", sampleErrorBacks)).toEqual(JN.makeBignum("1e30"));
+      expect(JN.fromString("1e140", sampleErrorBacks)).toEqual(JN.makeBignum("1e140"));
+
+      // for large bignums (> 1e140 ?), fromString() and makeBignum() can give structurally
+      // unequal results. so the following fail:
```

Comment:
@ds26gte why are these commented out? We should be testing to make sure they fail, right?